File: task4/task4_java/Wand.java
public class Wand extends Weapon {
    private static final int WAND_RANGE = 5;
    private static final int AXE_INIT_DAMAEG = 5;
    public Wand(Player owner){
        super(WAND_RANGE, AXE_INIT_DAMAEG, owner);
    }
    
    public void enhance(){
        this.effect += 5;
    }

    @Override
    public void action(int posx, int posy) {
        System.out.println("You are using Wand healing " + posx + " " + posy + ".");
        if (this.owner.pos.distance(posx, posy) <= this.range) {
            Player player = owner.game.getPlayer(posx, posy);
            if (player != null) {
                if (player instanceof Human && this.owner instanceof Human) {
                    player.increaseHealth(this.getEffect());
                    if (player.health > 80) {
                       player.health = 80; 
                    }
                } 
                else if (player instanceof Chark && this.owner instanceof Chark) {
                    player.increaseHealth(this.getEffect());
                    if (player.health > 100) {
                       player.health = 100; 
                    }
                } 
                else {
                    System.out.println("Can not use to heal the different race");
                }
            }
        }
        else {
            System.out.println("out of range");
        }
    }
}

File: task4/task4_python/Wand.py
from Weapon import *
#from Human import *
#from Chark import *
class Wand(Weapon):
    __WAND_RANGE = 5
    __WAND_INIT_DAMAGE = 5
    def __init__(self, owner):
        super(Wand, self).__init__(Wand.__WAND_RANGE, Wand.__WAND_INIT_DAMAGE, owner)
        self.maxhealth = owner._health
    
    def enhance(self):
        self._effect = self._effect + 5
    
    def action(self, posx, posy):
        print "You are using Wand healing " + str(posx) + " " + str(posy) + " " + "."
        if (self._owner._pos.distance(posx, posy) <= self._range):
            player = self._owner._game.getPlayer(posx, posy)
            if (player is None) == False:
                if isinstance(self._owner, type(player)):
                    player.increaseHealth(self.getEffect())
                    if player._health > self.maxhealth:
                        player._health = self.maxhealth
                else:
                    print "Cannot use to heal the different race"
        else:
            print "out of range"
